### PR TITLE
refactor(plot): refactor response time histogram plot

### PR DESCRIPTION
**Pull request by atsushi421**
## Description
This PR refactors the API to plot histograms of response times.
This PR changes the size of the figure as follows.

### Before
bokeh_plot

### After
bokeh_plot

## Related links
- JIRA ticket: https://tier4.atlassian.net/browse/T4PB-24807
- To reproduce: 
  - [ipynb, html](https://drive.google.com/drive/folders/1PDpG9JT_6J3Y6SUix7D625DZo7wydLnF?usp=share_link)
  - [trace data](https://drive.google.com/file/d/1xy1L4c-eGi9RSHog1qNgovTcwxoM74Z9/view?usp=share_link)
  - [architecture file](https://drive.google.com/file/d/1W9PDyj046xzapMA-p3o6HWtUhMA8JhIG/view?usp=share_link)

## Notes for reviewers
There is no degradation using CARET_report.
image

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/plot/histogram/histogram.py [hunk 1]
```diff
@@ -12,62 +12,51 @@
...
 # TODO: Migrate drawing process to visualize_lib
-class ResponseTimePlot(HistPlot):
+class ResponseTimeHistPlot(PlotBase):
 
     def __init__(
         self,
+        visualize_lib: VisualizeLibInterface,
         target: Sequence[Path],
         case: str = 'best-to-worst',
         binsize_ns: int = 10000000
```
Comment (rokamu623) [inline src/caret_analyze/plot/histogram/histogram.py:37]: [suggestion]
This function appears to be called only from `ResponseTimeHistPlotFactory.create_instanse`.
In that case, it does not seem necessary for the arguments to be optional.

```suggestion
    def __init__(
        self,
        visualize_lib: VisualizeLibInterface,
        target: Sequence[Path],
        case: str,
        binsize_ns: int
```

File: src/caret_analyze/plot/histogram/histogram_factory.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Sequence
```
Comment (rokamu623) [inline src/caret_analyze/plot/histogram/histogram_factory.py:17]: If `annotations` are to be introduced, I think `typing` is not necessary.

File: src/caret_analyze/plot/visualize_lib/bokeh/util/util.py [hunk 3]
```diff
@@ -77,7 +78,7 @@ def init_figure(
 
     return figure(
         frame_height=270, frame_width=800, title=title, y_axis_label=y_axis_label or '',
-        x_axis_label=x_axis_label, tools=tools, active_scroll=active_scroll
+        x_axis_label=x_axis_label or x_axis_label_, tools=tools, active_scroll=active_scroll
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/util/util.py:81]: This description is not intuitive.
Is it possible to implement the following? (in Line65)

```python
if x_axis_label is None:
  if x_axis_type == 'system_time':
    x_axis_label = 'system time [s]'
  elif xaxis_type == 'sim_time':
    x_axis_label = 'simulation time [s]'
  else:
    x_axis_label = xaxis_type

...

x_axis_label=x_axis_label, tools=tools, active_scroll=active_scroll

**Review by rokamu623 — CHANGES_REQUESTED** [review]
@atsushi421 
I have reviewed it.
Note that [suggestion] does not have to be modified.

File: src/caret_analyze/plot/visualize_lib/bokeh/util/util.py (hunk 3 above)
Comment (atsushi421) [reply src/caret_analyze/plot/visualize_lib/bokeh/util/util.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/289/commits/478cc692684d3e55acdb70a5b6626116c9f72d66

File: src/caret_analyze/plot/histogram/histogram_factory.py (hunk 2 above)
Comment (atsushi421) [reply src/caret_analyze/plot/histogram/histogram_factory.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/289/commits/14681c2f0479dce989b519d04afd33530227d5d7

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Comment by atsushi421** [comment]
@takeshi-iwanari 
Could you please review this PR?

**Review by takeshi-iwanari — APPROVED** [review]
LGTM

**Comment by atsushi421** [comment]
@takeshi-iwanari 
Thank you for your review!
By the way, is there any demand to get response time information in dataframe format?
(`to_dataframe()` method that are not implemented in this PR)

**Pull request merged**